include/circt/Dialect/HW/HWStructure.td

```diff
@@ -113,6 +113,14 @@ def HWModuleOp : HWModuleOpBase<"module",
     /// rewrites the output op to return the associated values.
     void appendOutputs(ArrayRef<std::pair<StringAttr, Value>> outputs);
 
+    /// Append a single input to the port list of the module.
+    /// Return the value of the associated block argument.
+    Value appendInput(Type type, const Twine &name);
```

Comment:
I don't think you want  Twine by reference, Twine is intended as a value type.

Reply:
Why are these different from the Output versions?  taking an array ref is only two more characters at the callsite for a single port append.

Reply:
> I don't think you want Twine by reference, Twine is intended as a value type.

Eep, FWIW Twine's [are suggested to be const ref in arguments](https://llvm.org/doxygen/classllvm_1_1Twine.html):

> A [Twine](https://llvm.org/doxygen/classllvm_1_1Twine.html) is not intended for use directly and should not be stored, its implementation relies on the ability to store pointers to temporary stack objects which may be deallocated at the end of a statement. Twines should only be used accepted as const references in arguments, when an API wishes to accept possibly-concatenated strings.

I think this is required to preserve the lifetimes of the temporaries used.

An rvalue reference (`Twine &&`) does that too I believe, but probably better to stick with const refs for now (fun fact: according to a quick grep, MLIR has the only instance of `Twine &&` LLVM-wide, an overload that also takes `const Twine &`, wonder what that's about/for? :thinking:).

Reply:
They're different from the array version since they also return a value associated with the newly added argument.